sklearn/utils/testing.py

```diff
@@ -300,47 +304,44 @@ def ignore_warnings(obj=None):
 
     >>> ignore_warnings(nasty_warn)()
     42
-
     """
     if callable(obj):
-        return _ignore_warnings(obj)
+        return _IgnoreWarnings(category=category)(obj)
     else:
-        return _IgnoreWarnings()
-
-
-def _ignore_warnings(fn):
-    """Decorator to catch and hide warnings without visual nesting"""
-    @wraps(fn)
-    def wrapper(*args, **kwargs):
-        # very important to avoid uncontrolled state propagation
-        clean_warning_registry()
-        with warnings.catch_warnings(record=True) as w:
-            warnings.simplefilter('always')
-            return fn(*args, **kwargs)
-            w[:] = []
-
-    return wrapper
+        return _IgnoreWarnings(category=category)
 
 
 class _IgnoreWarnings(object):
+    """Improved and simplified Python warnings context manager and decorator.
 
-    """Improved and simplified Python warnings context manager
-
+    This class allows to ignore the warnings raise by a function.
     Copied from Python 2.7.5 and modified as required.
```

Comment:
This is no longer a straigth copy from Python 2.7.5. This part of the docstring needs to be updated.